pkgs/development/tools/build-managers/ninja/default.nix

```
@@ -1,39 +1,67 @@
-{ lib, stdenv, fetchFromGitHub, fetchpatch, python3, buildDocs ? true, asciidoc, docbook_xml_dtd_45, docbook_xsl, libxslt, re2c }:
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchpatch
+, asciidoc
+, docbook_xml_dtd_45
+, docbook_xsl
+, libxslt
+, python3
+, re2c
+, buildDocs ? true
+}:
 
-with lib;
-
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "ninja";
   version = "1.11.1";
 
   src = fetchFromGitHub {
     owner = "ninja-build";
     repo = "ninja";
-    rev = "v${version}";
-    sha256 = "sha256-LvV/Fi2ARXBkfyA1paCRmLUwCh/rTyz+tGMg2/qEepI=";
+    rev = "v${finalAttrs.version}";
```

```suggestion
    rev = "v${version}";
```
that is not working because of the fixed output derivation

---

@roberth ping as a new bad case

---

> not working

You _can_ replace `rec` like this.

I suppose you expect to be able to override both `rev` and `hash` in one go? `finalAttrs` would make this work if `fetchFromGitHub` had a `.override`:

```nix
(ninja.overrideAttrs(_: { version = "9000"; })
).overrideAttrs(old: {
  src = super.src.override (_: { hash = "foo"; });
})
```

It's DRY in terms of the packaging domain, but overriding the "same" derivation twice is a little excessive.

This could be prevented by adding `hash` to the `ninja` derivation directly, but that derivation shouldn't have `hash`, as the `ninja` builder doesn't (and probably shouldn't) use it.
For it to work in one `overrideAttrs`, or perhaps one `overrideAttrs` per derivation (`ninja`, `source`), we'd need further changes, and I don't have a single answer for what those should be.

While we might do slightly better than the status quo, overriding is never going to be really pretty, because it is fundamentally coupled with the implementation of whatever you're overriding.

---

I am being more modest here and merely getting rid of `rec`.

I will try another time.